src/silo/storage/database_partition.cpp

```diff
@@ -124,6 +124,13 @@ void DatabasePartition::validateMetadataColumns() const {
          );
       }
    }
+   for (const auto& col : columns.bool_columns) {
```

Comment:
Regarding cognitive complexity: 
* Either we ignore it here (we unfortunately also did this in many other places
* Or (my preferred way) we extract methods for every `for` in this method (seems easy enough on a quick glimpse).